### PR TITLE
fix(#653): strengthen assert_file_order_in_html against substring false positives

### LINKED ISSUE #653: test: strengthen assert_file_order_in_html against structural false positives
## Summary

`assert_file_order_in_html` in `tests/test_helpers.py` verifies files appear in order by substring search, but could produce false positives if one filename is a substring of another (e.g., `file.txt` matching inside `my-file.txt`).

## Issue

The helper uses `str.index()` for ordering checks. If filenames share common substrings, the search may match the wrong occurrence, giving a false-positive ordering result.

## Proposed Fix

- Use word-boundary aware search or require exact token matches when validating file ordering
- Or add a note in the docstring warning about this limitation

## Files Affected

- `tests/test_helpers.py` — `assert_file_order_in_html` function

## References

- PR #652 (refactor: consolidate web route test helpers and fixtures)
- Copilot review comment on PR #652, comment ID 2899854261

## CCPM Tracking

**Epic**: testing-qa
**Priority**: Low
**Effort Estimate**: 1-2 hours

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary

- Replaces `str.index()` in `assert_file_order_in_html` with regex exact-token matching using negative lookahead/lookbehind on filename boundary chars (`[.\w-]`), so `file.txt` cannot accidentally match inside `my-file.txt`
- Private helper `_find_filename_in_html` uses `re.compile().search(pos=start)` — no string-slice allocation per call
- Adds `tests/test_test_helpers.py` (19 tests) covering all 4 helpers, including the false-positive substring case that motivated the issue
- Excludes `.claude/` from ruff (pre-existing internal tooling violations unrelated to project lint baseline)

## Test plan

- [ ] `test_no_false_positive_substring_match` — `file.txt` must not match in `<li>my-file.txt</li>`
- [ ] `test_exact_token_matches_when_present` — both short and long names work when both present
- [ ] `test_prefix_name_does_not_match_longer_name` — `report` not found in `report_final.txt`
- [ ] All 19 helper tests pass
- [ ] Smoke suite passes (178 tests)
- [ ] CI passes

Closes #653

## Summary by CodeRabbit

* **Chores**
  * Updated lint configuration to exclude an additional directory from analysis.

* **Refactor**
  * Improved test assertion helpers to perform exact token matching and stricter ordering checks; error messages clarified for missing matches.

* **Tests**
  * Added comprehensive unit tests covering assertion helpers, ordering, matching edge cases, duplicates, and failure modes.

### CODE REVIEW

File: tests/test_helpers.py [hunk 1]
```diff
@@ -2,31 +2,64 @@
...
+            current_index = _find_filename_in_html(text_lower, file_lower, previous_index + 1)
         except ValueError as err:
-            raise AssertionError(f"File {file_lower} not found in response") from err
+            raise AssertionError(f"File '{file_lower}' not found as exact token in response") from err
 
         assert current_index > previous_index, (
-            f"Files not in correct order: {file_lower} at index {current_index} "
+            f"Files not in correct order: '{file_lower}' at index {current_index} "
             f"should come after previous at {previous_index}"
         )
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/test_helpers.py]: ✅ Fixed: removed the assertion. As noted, compiled.search(text, pos=previous_index+1) guarantees match.start() >= previous_index+1 > previous_index, so the assertion can never fire. Ordering is enforced by the pos parameter.

**Comment by curdriceaurora** [comment]
@coderabbitai review
